### PR TITLE
Fix recent build error on ppc64le 

### LINKED ISSUE #128130: PyTorch fails to build on ppc64le
### 🐛 Describe the bug

I am getting an error while building pytorch for ppc64le
The error is : 
/home/manjulmohan/pytorch_may17/pytorch-3/aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:735:3:   required from here
/home/manjulmohan/pytorch_may17/pytorch-3/aten/src/ATen/cpu/vec/vec_base.h:632:6: error: use of deleted function ‘double& at::vec::DEFAULT::Vectorized<double>::operator[](int)’
  632 |     c[i] = a[i] * b[i];

The error is caused by the commit id: 457b9f7397c83953e7f3124f02738427aaff61cd

### Versions

python collect_env.py
Collecting environment information...

Core(s) per socket:  12
Socket(s):           4
NUMA node(s):        4
Model:               2.0 (pvr 0080 0200)
Model name:          POWER10 (architected), altivec supported
Hypervisor vendor:   pHyp
Virtualization type: para
L1d cache:           32K
L1i cache:           48K
L2 cache:            1024K
L3 cache:            4096K
NUMA node0 CPU(s):   0-95
NUMA node1 CPU(s):   96-191
NUMA node2 CPU(s):   192-287
NUMA node3 CPU(s):   288-383

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] optree==0.11.0
[pip3] torch==2.4.0a0+git6c503f1
[conda] nomkl                     3.0                           0    rocketce
[conda] numpy                     1.26.4          py311hbc68021_0    conda-forge
[conda] optree                    0.11.0                   pypi_0    pypi
[conda] torch                     2.4.0a0+git6c503f1          pypi_0    pypi

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @malfet @seemethere

### PR DESCRIPTION
**Pull request by pratiklp00**
This PR will fix the recent build issue observed on ppc64le. 
Fixes #128130

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/FlashAttentionKernel.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@ inline void _scale_attn_mask_fusion_kernel(
   constexpr int64_t T2_n = 1;
   auto vec_scale = at::vec::VectorizedN<T1, T1_n>(val);
   int64_t i = 0;
+ #if !defined(__powerpc__)
```
Comment (albanD) [inline aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:41]: won't think make this code silently wrong though?

Comment (pratiklp00) [reply aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:41]: This is temporary fix.

Comment (ChipKerchner) [reply aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:41]: Could you explain why you think this PR will make it `silently wrong`?

Looks like the change will make PowerPC use only scalar operations and skip the vector operations.  The end result is the PPC will just be slower for this function but still correct.

Comment (albanD) [reply aten/src/ATen/native/cpu/FlashAttentionKernel.cpp:41]: Ho sorry, I didn't realized the loop below continues on the same i and not only does the tail. Sorry about that!

**Review by albanD — APPROVED** [review]
Sounds good!

**Comment by pratiklp00** [comment]
Hi @albanD I have committed the new changes which will be the proper fix for the issue #128130 could you please review it. Thanks.

**Review by albanD — APPROVED** [review]
Ho super cool, thanks for taking the time to do this!

**Comment by pratiklp00** [comment]
Hi @albanD could you please help me with merge

**Comment by pratiklp00** [comment]
@malfet @soulitzer Could you please help me with merge.

**Review by malfet — APPROVED** [review]

**Comment by pratiklp00** [comment]
@albanD @malfet @@soulitzer After rebase there are no CI test failure. Sorry for inconvenience can you please try merging it again.

**Comment by pratiklp00** [comment]
@malfet @albanD Thanks for the review and help. I would like to know if there is way to push this changes to release/2.4 branch.

**Comment by malfet** [comment]
@pratiklp00 please read the instructions in https://github.com/pytorch/pytorch/issues/132400 and add appropriate comment if it qualifies a cherry-pick criteria

**Comment by pratiklp00** [comment]
@malfet Thank you.

**Pull request closed without merging**